framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -266,6 +278,69 @@ MeshDiagnosticsGenerator::checkSidesetsOrientation(const std::unique_ptr<MeshBas
   }
 }
 
+void
+MeshDiagnosticsGenerator::checkWaterTightSidesets(const std::unique_ptr<MeshBase> & mesh) const
+{
+  /*
+  Algorithm Overview:
+  1) Loop through all elements
+  2) For each element loop through it's edges
```

Comment:
```suggestion
  2) For each element loop through all its sides
```
edge is a side of a 2D element, it s necessarily 1D